cv-data-controller/src/main/java/com/trihydro/cvdatacontroller/controller/MilepostController.java

```diff
@@ -308,6 +311,74 @@ public ResponseEntity<Collection<com.trihydro.cvdatacontroller.model.Milepost>>
 		return ResponseEntity.ok(data);
 	}
 
+	@RequestMapping(method = RequestMethod.POST, value="/set-milepost-cache")
+	public ResponseEntity<String> setMilepostCache(@RequestBody SetMilepostCacheRequest milepostCacheBody) {
+		
+		utility.logWithDate("Setting milepost cache for timID: " + milepostCacheBody.getTimID());
+
+		if (milepostCacheBody.getMileposts().size() == 0 || milepostCacheBody.getTimID() == null) {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("Invalid Request: please provide a valid milepost list and timID");
+		}
+		if (milepostCache.containsKey(milepostCacheBody.getTimID())) {
```

Comment:
question: if we happen to want to update the cache for say an extended workzone, should this just replace the existing entry instead of failing?

Reply:
Yes, I think it makes more sense to replace the existing entry instead of failing. This has been updated.